src/annotator/sidebar.js

```diff
@@ -48,48 +44,58 @@ function createSidebarIframe(config) {
   return sidebarFrame;
 }
 
+const defaultConfig = {};
+
 /**
  * The `Sidebar` class creates the sidebar application iframe and its container,
  * as well as the adjacent controls.
  */
 export default class Sidebar extends Guest {
+  /**
+   * @param {HTMLElement} element
+   * @param {Record<string, any>} config
+   */
   constructor(element, config) {
-    if (config.theme === 'clean' || config.externalContainerSelector) {
-      delete config.pluginClasses.BucketBar;
-    }
-
-    let externalContainer = null;
-
-    if (config.externalContainerSelector) {
-      externalContainer = document.querySelector(
-        config.externalContainerSelector
-      );
-    }
+    super(element, { ...defaultConfig, ...config });
 
     let externalFrame;
     let frame;
 
-    if (externalContainer) {
-      externalFrame = externalContainer;
+    if (config.externalContainerSelector) {
+      externalFrame =
+        document.querySelector(config.externalContainerSelector) || element;
     } else {
       frame = document.createElement('div');
       frame.style.display = 'none';
       frame.className = 'annotator-frame annotator-outer';
 
       if (config.theme === 'clean') {
         frame.classList.add('annotator-frame--theme-clean');
+      } else {
+        // The instantiation of the BucketBar plugin can't be done as other plugin.
+        // The sidebar must be inserted inside a shadow DOM which is not accessible
+        // via a common querySelector.
```

Comment:
I understand what this comment means but I think it may not make sense to someone in future who wasn't aware of the way the code used to work.

I think it would be better to say that the `BucketBar` is a "plugin" for legacy reasons and is now constructed here so that the parent element can be passed into the constructor.

Reply:
Done

Reply:
Done